### PR TITLE
Automatic labelling

### LINKED ISSUE #42: Automatic labelling
@lukashergt latest fix for figure legends #30 means that we now have matplotlib-like handles for creating figure legends. However, these are only applied if you supply a label keyword argument to the plot. It might be better if this was automatically generated/could be set as a member variable for MCMCSamples, so that users didn't need to remember to supply labels to generate plots.

### PR DESCRIPTION
**Pull request by williamjameshandley**
# Description

Added a `label` attribute to `MCMCSamples`, which provides a default for the plots. Labels are defined to be the basename of the root if loaded from file, None otherwise.

Fixes #42

# Checklist:

- [x] I have performed a self-review of my own code
- [x] My code is PEP8 compliant (`flake8 anesthetic tests`)
- [x] My code contains compliant docstrings (`pydocstyle --convention=numpy anesthetic`)
- [x] New and existing unit tests pass locally with my changes (`python -m pytest`)
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: anesthetic/samples.py [hunk 1]
```diff
@@ -43,32 +44,40 @@ class MCMCSamples(WeightedDataFrame):
...
+        mapping from columns to prior limits
+
+    label: str
+        Legend label
 
     """
 
-    _metadata = WeightedDataFrame._metadata + ['tex', 'limits', 'u', 'root']
+    _metadata = WeightedDataFrame._metadata + ['tex', 'limits', 'u',
+                                               'root', 'label']
```
Comment (lukashergt) [inline anesthetic/samples.py]: Not really for this PR, but as a side remark: `'u'` is very cryptic. I've gone through the code once to figure out what it does, but I forgot again... Is there maybe a more descriptive name for that variable?

Similar for `w`. Although, not as bad as it often appears with the key `'weight'` (and at least to me it's somewhat familiar/expected)...

File: tests/test_samples.py [hunk 2]
```diff
@@ -187,9 +206,10 @@ def test_plot_2d_legend():
                     assert(all([isinstance(h, Rectangle) for h in handles]))
     plt.close('all')
 
+    # Test label kwarg label kwarg for hist and scatter
```
Comment (lukashergt) [inline tests/test_samples.py]: `label kwarg` repeated

File: anesthetic/samples.py (hunk 1 above)
Comment (williamjameshandley) [reply anesthetic/samples.py]: OK, I've added an issue #44 for further discussion and to remind us to address this at some point

File: tests/test_samples.py (hunk 2 above)
Comment (williamjameshandley) [reply tests/test_samples.py]: Fixed in 9f457a2

**Pull request merged**